docs/dev/src/design/keys.md

```diff
@@ -23,15 +42,39 @@ The record corresponding to key `(0, 1)` has been inserted with `(0, 1, 2, 3)`.
 It may not be the minimal set of columns required to identify a record,
 for instance `group key` could be part of the stream key, to specify the distribution of records.
 
-## Primary Key (Storage)
+## Distribution key
+
+This key specifies the distribution of the data streams, as mentioned in [Consistent Hash - Streaming](../design/consistent-hash.md#streaming).
+We want data to be distributed in a way that minimizes data skew,
+and maximizes data locality, for more efficient stateful processing.
+
+To ensure data consistency of updates (U-, U+),
```

Comment:
It doesn't have to be `U-` `U+`. The order of updates in any form for records with the same stream key should be preserved.